src/test/rustdoc/issue-74134-4.rs

```diff
@@ -0,0 +1,11 @@
+// compile-flags: --document-private-items
+#![deny(intra_doc_link_resolution_failure)]
+
+// Linking from a public item to a private type is fine with --document-private-items.
```

Comment:
This seems a little inconsistent to me ... the warning is 'magically' going away because we added an extra flag (or in other words, you can _only_ document with `--document-private-items` if you want to avoid warnings). I think it would be best to warn in both cases, but if it's tricky to implement we can let it go, it's being more permissive after all.

Reply:
I just tried to come up with use-cases in which you always run with `--document-private-items` anyway and don't want to see any warnings. But then you can do `#![allow(intra_doc_link_resolution_failure)]` of course. So yes, maybe such links should always warn.

But I'd say that is outside the scope of this PR?

Reply:
Hmm, I just checked and you're right that this behavior is also present on nightly without your changes. So no need to change it here, might be nice to add a `// FIXME` though.

Reply:
It seems actually rather trivial to fix: https://github.com/dennis-hamester/rust/commit/e8d168986895da823245213e47e0ca2e652eef89

I still feel like it doesn't fit in this PR, as it changes behavior beyond what is required to fix #74134. But I can squeeze it in of course if you want. It won't confuse bors, will it?

Reply:
Let's make it a separate PR, you've convinced me it's a separate issue :)

Reply:
I'll open one after this hits master.